### PR TITLE
4844: Fix Dencun e2e tests

### PR DESCRIPTION
**Pull request by danyalprout**
**Description**

Currently we [pass `nil` for the `L1BlobFetcher`](https://github.com/ethereum-optimism/optimism/blob/23dd85a41d7c2e029da87861716572916d325d40/op-e2e/actions/l2_verifier.go#L62), which causes [this error](https://github.com/ethereum-optimism/optimism/blob/23dd85a41d7c2e029da87861716572916d325d40/op-node/rollup/derive/data_source.go#L51) to occur for post-Ecotone blocks which results in `ActL2PipelineFull` never returning as the pipeline is never set to idle (due to it being an supported error).

This PR passes makes the following changes:

* Provide a no-op mock for fetching blobs
  *  Should never be called as the actions tests have their own [batcher](https://github.com/ethereum-optimism/optimism/blob/23dd85a41d7c2e029da87861716572916d325d40/op-e2e/actions/l2_batcher.go#L81) which only supports calldata.
* Fail the tests if there's an unexpected error

### CODE REVIEW

File: op-e2e/actions/dencun_fork_test.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func TestDencunL1ForkAtGenesis(gt *testing.T) {
 	require.Equal(t, sequencer.SyncStatus().UnsafeL2, verifier.SyncStatus().UnsafeL2, "verifier and sequencer agree")
 }
 
-func TestDencunL2ForkAfterGenesis(gt *testing.T) {
+func TestDencunL2ForkAtGenesis(gt *testing.T) {
```
Comment (axelKingsley) [inline op-e2e/actions/dencun_fork_test.go]: I notice we are swapping the names of these tests, "AtGenesis" becomes "AfterGenesis" and vice-versa. But looking a the content of the tests, I'm not sure if that was intentional?

What is now called "TestDencunL2ForkAtGenesis" references "crossing the fork boundary" and contains 1 extra call to the miner than "TestDencunL2ForkAfterGenesis"

File: op-e2e/actions/l2_sequencer.go [hunk 2]
```diff
@@ -28,6 +28,18 @@ func (m *MockL1OriginSelector) FindL1Origin(ctx context.Context, l2Head eth.L2Bl
 	return m.actual.FindL1Origin(ctx, l2Head)
 }
 
+// emptyL1BlobsFetcher is a no-op blobs provider .The actions test batcher currently only supports using calldata.
```
Comment (axelKingsley) [inline op-e2e/actions/l2_sequencer.go]: "provider .The" -> "provider. The"

But also, could you rephrase this comment somewhat? I think it means that this struct is used by e2e tests only, in order to confirm that blobs aren't used pre-dencun, but I'm not sure.

File: op-e2e/actions/l2_sequencer.go [hunk 3]
```diff
@@ -28,6 +28,18 @@ func (m *MockL1OriginSelector) FindL1Origin(ctx context.Context, l2Head eth.L2Bl
...
+type emptyL1BlobsFetcher struct {
+	t Testing
+}
+
+var _ derive.L1BlobsFetcher = &emptyL1BlobsFetcher{}
+
+func (e *emptyL1BlobsFetcher) GetBlobs(ctx context.Context, ref eth.L1BlockRef, hashes []eth.IndexedBlobHash) ([]*eth.Blob, error) {
+	e.t.Fatal("actions test do not support blobs")
+	return []*eth.Blob{}, nil
```
Comment (axelKingsley) [inline op-e2e/actions/l2_sequencer.go]: nit: I think you could return `nil, nil` here

File: op-e2e/actions/dencun_fork_test.go (hunk 1 above)
Comment (danyalprout) [reply op-e2e/actions/dencun_fork_test.go]: Thanks! I swapped them as the AfterGenesis `offset` was 0 and the AtGenesis was 24 -- which seemed the opposite way around. 

Just pushed a commit to rework (and relocate) these tests.

**Comment by protolambda** [comment]
Rebased on `dencun` branch, moved back tests in original `dencun` place for more readable diff. Changes look good. Added a few minor fixes (set regolith offset, deconflict with develop-branch blob-fetcher, etc.).
